torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -3489,6 +3489,160 @@ Tensor i1e_backward(
   });
 }
 
+// lu_solve is a map (LU, B) -> (PLU)^{-1} B,
+// where LU = L + U - I and P is a permutation matrix.
+// Note that the map is not differentiable wrt P!
+//
+// Let 1 = ones_like(LU),
+// 1_U = 1.triu(),
+// 1_L = 1 - 1_U (note the zero diagonal),
+// * := the Hadamard (element-wise) product
+//
+// Forward AD:
+//
+// Let X := U^{-1} L^{-1} P^T B be the output of the function.
+// Also, the LU input of the function could be represented as
+// LU = L + U - I.
+//
+// Differentiating LU = L + U - I produces:
+// dLU = dL + dU.
+// Noting that dL and dU are lower- and upper-triangular, respectively,
+// and that the diagonal of L is never explicitly exposed, so
+// diag(dL) = 0, it follows
+// dL = dLU * 1_L,
+// dU = dLU * 1_U.
+//
+// Differentiating X = U^{-1} L^{-1} P^T B produces:
+// dX = dU^{-1} L^{-1} P^T B + U^{-1} dL^{-1} P^T B + U^{-1} L^{-1} P^T dB
```

Comment:
They are the same. Proof:
```
A A^{-1} = I => dA A^{-1} + A d(A^{-1}) = 0 =>
dA = - A d(A^{-1}) A,
d(A^{-1}) = - A^{-1} dA A^{-1}.

Then we get, from definition of dA:
(dA)^{-1} = - A^{-1} (dA^{-1)^{-1} A^{-1},

then
dA (dA^{-1}) = [ - A d(A^{-1}) A] [ - A^{-1} (dA^{-1))^{-1} A^{-1}] = d(A^{-1}) (dA^{-1})^{-1} = I.
Since d(A}) is arbitrary (not necessarily an identity), it follows, from the uniqueness of inverse that
d(A^{-1}) = (dA)^{-1}

```

Reply:
Why is that? It is an operator `d` applied to `A^{-1}`. We never consider `(dA)^{-1}`, there I would have introduces parentheses explicitly.

Reply:
Yeah, that's the thing, the precedence is not clear.

Reply:
It is clear as it is consistent, plus function notation makes it clear, otherwise the parentheses will bloat the description. There are plenty of inverses.

Reply:
It is not clear for some people I'd say it's not clear :D
For what is worth, in differential geometry the differential does have preference over virtually any other operation.